javascript/app-event-analysis/src/components/CounterfactualLines.hooks.ts

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import { isEmpty } from 'lodash'
+import { useMemo } from 'react'
+
+import type { D3ScaleLinear, OutputData, PlaceboOutputData } from '../types.js'
+import { useLineColors } from '../utils/useColors.js'
+import type {
+	CounterFactualLineData,
+	CounterfactualLinesProps,
+} from './CounterfactualLines.types.js'
+
+const CONTROL_LINE_WIDTH = 2
+
+function useShowCounterfactualOnly(
+	applyIntercept: boolean,
+	showSynthControl: boolean,
+) {
+	return useMemo(
+		() => showSynthControl && !applyIntercept,
+		[showSynthControl, applyIntercept],
+	)
+}
+
+function useShowTreatedAndCounterfactual(x1: number) {
+	return useMemo(() => x1 > 0, [x1])
+}
+
+function useShowControlOnly(
+	x1: number,
+	applyIntercept: boolean,
+	showSynthControl: boolean,
+) {
+	const showCounterfactualOnly = useShowCounterfactualOnly(
+		applyIntercept,
+		showSynthControl,
+	)
+	const showTreatedAndCounterfactual = useShowTreatedAndCounterfactual(x1)
+	return useMemo(
+		() => showCounterfactualOnly && showTreatedAndCounterfactual,
+		[showCounterfactualOnly, showTreatedAndCounterfactual],
+	)
+}
+
+function useOutputDataNonPlacebo(
+	hoverUnit: string | undefined,
+	outputData: (OutputData | PlaceboOutputData)[],
+): OutputData {
+	const outputIndex = outputData.findIndex(output =>
+		hoverUnit?.includes(output.treatedUnit),
+	)
+	const outputDataNonPlacebo = outputData[outputIndex] as OutputData
+	return useMemo(() => outputDataNonPlacebo, [outputDataNonPlacebo])
+}
+
+function useX1(
+	outputDataNonPlacebo: OutputData,
+	xScale: D3ScaleLinear,
+): number {
+	return useMemo(
+		() =>
+			outputDataNonPlacebo &&
```

Comment:
`outputDataNonPlacebo?.time_before_intervention != null`

* The `?` lets you skip the initial check.
* Using `!= null` or `== null` is generally preferred to `!== undefined` or `== undefined`. It will catch both `null` and `undefined`. This is the one place where double-equal is preferred in JavaScript.